tests/phpunit/includes/TestCase.php

```diff
@@ -11,15 +11,38 @@
 namespace Google\Site_Kit\Tests;
 
 use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Util\Feature_Flags;
 use Google\Site_Kit\Core\Util\Input;
+use Google\Site_Kit\Core\Util\JSON_File;
 use Google\Site_Kit\Tests\Exception\RedirectException;
 use PHPUnit_Framework_MockObject_MockObject;
 
 class TestCase extends \WP_UnitTestCase {
-
 	// Do not preserve global state since it doesn't support closures within globals.
 	protected $preserveGlobalState = false;
 
+	protected static $featureFlagsConfig;
+
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+
+		if ( ! self::$featureFlagsConfig ) {
+			self::$featureFlagsConfig = new JSON_File( GOOGLESITEKIT_PLUGIN_DIR_PATH . 'feature-flags.json' );
+		}
+
+		self::reset_feature_flags();
+	}
+
+	public static function tearDownAfterClass() {
+		parent::tearDownAfterClass();
+		self::reset_feature_flags();
+	}
```

Comment:
I chose these so that classes could do their own setup/teardown for each test but it wouldn't leak between test cases. The JSON file is only instantiated and read once as well.